frontend/src/core/comments/components/AddComment.js

```diff
@@ -26,72 +46,331 @@ export default function AddComments(props: Props) {
+            const el = ref.current;
+            const domRange = ReactEditor.toDOMRange(editor, target);
+            const rect = domRange.getBoundingClientRect();
+            const teamCommentsEl = document.querySelector('.top');
+            const teamCommentsRect = !teamCommentsEl ? null : teamCommentsEl.getBoundingClientRect();
+            const teamCommentsActive = !teamCommentsEl ? false : teamCommentsEl.contains(document.activeElement);
+            const translateCommentsEl = document.querySelector('.history');
+            const translateCommentsRect = !translateCommentsEl ? null : translateCommentsEl.getBoundingClientRect();
+            const translateCommentsActive = !translateCommentsEl ? false : 
+                translateCommentsEl.contains(document.activeElement);
+
+            let setTop = (rect.top + window.pageYOffset) + 21;
+            let setLeft = rect.left + window.pageXOffset;
+            
+            // If suggestions overflow the window or teams container height then adjust the
+            // position so they display above the comment
+            const suggestionsHeight = el.clientHeight + 10;
+            const teamCommentsOverflow = !teamCommentsRect ? false : 
+                ((setTop + el.clientHeight) - 51) > teamCommentsRect.height;
+
+            if ((teamCommentsActive && teamCommentsOverflow) || setTop + suggestionsHeight > window.innerHeight) { 
+                setTop = (setTop - suggestionsHeight) - 21;
+            }
+
+            // If suggestions in team comments scroll below or suggestions in translation
+            // comments scroll above the next section or overflow the window then hide the suggestions
+            if ((teamCommentsRect && teamCommentsActive && (((setTop + suggestionsHeight) - 61) > teamCommentsRect.height)) ||
+                (translateCommentsRect && translateCommentsActive && (rect.top < translateCommentsRect.top)) || 
+                (translateCommentsRect && translateCommentsActive && (setTop + suggestionsHeight > window.innerHeight))) {
+                el.style.display = 'none';
+            }
+            
+            // If suggestions overflow the window width in team comments or the right side of the 
+            // translations comments then adjust the position so they display to the left of the mention
+            const suggestionsWidth = el.clientWidth
+            const translateCommentsOverflow = !translateCommentsRect ? false :
+                setLeft + suggestionsWidth > translateCommentsRect.right;
+            
+            if (setLeft + suggestionsWidth > window.innerWidth || 
+                (translateCommentsActive && translateCommentsOverflow)) {
+                setLeft = rect.right - suggestionsWidth;
+            }
+
+            el.style.top = `${setTop}px`;
+            el.style.left = `${setLeft}px`;
+        }
+    }, [chars.length, editor, index, search, target, scrollPosition]);
+
+    React.useEffect(() => {
+       // $FLOW_IGNORE 
```

Comment:
I found the comment you linked to previously, and it does eliminate the Flow errors if used as written. However, as I understand it React uses `SyntheticEvents` and this is where the errors come in:
1. `SynthticEvent<>` does not exist on `EventListener`
1. There is no SynthticEventListener
1. Without `EventListener` typed on the event handler then I get errors on all of the listeners re: `Event` being incompatible with `SynthticEvent` 

So, the below code eliminates the Flow errors but uses `Event` in place of the `SyntheticEvent`.  Is that OK, or should I continue to ignore flow and add a comment?
```js
        const handleScroll: EventListener = (e: Event) => {
            if (e.currentTarget instanceof HTMLElement) {
                const element = e.currentTarget;
                setScrollPosition(element.scrollTop);
            }
        }

        const historyScroll: HTMLElement | null = document.querySelector('#history-list');
        const teamsScroll: HTMLElement | null = document.querySelector('#react-tabs-3');
```

Reply:
Thanks for looking into it. I think we should be using the SyntheticEvent wrapper, since it works the same way across browsers. We use it everywhere else.